.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,100 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+name: Documentation
+
+on: 
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
+  links-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Set Up Env
+        run: make env
+      
+      - name: Install pandoc
+        run: |
+          sudo apt-get update
+          sudo apt-get install pandoc
+
+      - name: Build docs
+        run: make docs
+
+      # NOTE: 
+      #   I was not able to make "lycheeverse/lychee-action@v1.2.0" work, 
+      #   My main problem was that the glob patterns (e.g. "./docs/build/html/**/*.html") did not work.
+      #
+      #   In order to use the glob pattern properly we need to surround it with double quotes (") but 
+      #   when we do this, some part of the logic that handles the 'with.args' key, additionally wraps 
+      #   it into ordinary quotes ('), as a result lychee binary is executed as - lychee '"glob/**/*.pattern"'
+      #
+      #   I decided to overcome this with help of a custom script (take a look at the job step below)
+      #
+      #   Also, please take a look at this two related issues in the lychee-action repo
+      #   - https://github.com/lycheeverse/lychee-action/issues/68
+      #   - https://github.com/lycheeverse/lychee-action/issues/67
+      #
+      # - name: Link Checker
+      #   uses: lycheeverse/lychee-action@v1.2.0
+      #   with:
+      #     args: --exclude-file docs/.lycheeignore --verbose --exclude-loopback --exclude-mail --no-progress ./docs/build/html/**/*.html
+      #     fail: true
+      #   env:
+      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+      - name: Link Checker
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        run: |
+          set -uxo pipefail
+          
+          curl \
+            -L --output lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz \
+            https://github.com/lycheeverse/lychee/releases/download/v0.8.2/lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz
+          
+          tar -xvzf lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz
+
+          echo "$GITHUB_TOKEN"
+
+          ./lychee \
+            README.rst FAQ.rst "./docs/build/html/**/*.html" \
+            --base ./docs/build/html \
```

Comment:
Why are we checking this on the html and not on the rst? this means that every pr we need to run make docs before testing lychee?

Reply:
do you mean, why we are not checking docs source code?

good question, I also was thinking about it/ should we check docs source code or the result of the build.
I think the last one is better because a lot of stuff is expanded during the build, like interlinks between different parts of the documentation, etc.

I think we can cache a result of the docs build ('build' folder), and then use it during the next build, but this question needs additional exploration. But, it will make workflow more complicated - not sure if it is worth/ Maybe, let's run it only on the release?

Reply:
Got it. in addition - can it also validate links in the README?

Reply:
yes, it checks FAQ README and all generated html files

```
README.rst FAQ.rst "./docs/build/html/**/*.html"
```

Reply:
We have some more readme (For example inside the examples/ folder, or under suites/. I think it's better to find all rst files (which are not under docs/) and validate them

Reply:
yup, You are right, I changed the glob pattern to "./**/*.rst"